### DISCUSSION TITLE
Addition of incorrect layer labels when using `subject2mni` on segmented images

### DISCUSSION THREAD

**Discussion by KTZ228** [simnibs/simnibs#175] [7.Creating head models] [answered]
I want to transform the output from charm (the 3D matrix consisting of layer labels) to MNI space for further processing.
The issue that I have run into is that when I use the `subject2mni` function, an additional layer of incorrect labels seem to be added along the border of the tissue as can be seen when comparing the figures below.
Does anyone know what the cause of this is?

This shows what my brain looked like before the transformation. 
*Layers 1 and 2 are merged here since it simplifies our acoustic calculations.
![Screenshot 2023-07-26 114337](https://github.com/simnibs/simnibs/assets/51954604/d9bcf9d2-a3fe-4c81-b2a5-ba5abd8920c9)

And here you can clearly see the addition of a layer consisting of labels 1 and 2 (neural tissue) on the borders of the skin and neural tissue. *The coordinates of these two are not perfectly aligned but the layer mentioned is present and equally thin everywhere.
![Screenshot 2023-07-26 114401](https://github.com/simnibs/simnibs/assets/51954604/8991919a-57aa-4fb1-ad14-5c6351f90d12)

**Comment by oulap** [comment, marked as answer]
subject2mni uses linear interpolation as default and the lines you see are interpolation artifacts due to this.

Try passing the following flag to the call:

--interpolation_order 0

that switches the interpolation to nearest neighbor and the weird lines should be gone

**Reply by KTZ228** [reply]
Just managed to check. This was indeed the issue, thank you so much for your help.
Would I be correct in assuming that it would be best to leave this setting at default for anatomical scans and change it to 0 for any other non-anatomical files?

**Reply by oulap** [reply]
For labeled data, i.e., segmentations or masks, use 0 but for continuous data, i.e., MRI scan or electric field magnitude, use 1.

**Reply by KTZ228** [reply]
That is a better way to phrase it but glad we agree
